controllers/concept.go

```diff
@@ -148,3 +152,147 @@ func (u ConceptController) RetrieveBreakdownStatsBySourceIdAndCohortIdAndConcept
+	for _, concept := range sortedConcepts {
+		row = append(row, strconv.Itoa(conceptToPeopleCount[concept]))
+	}
+
+	return row
+}
+
+func (u ConceptController) getFilteredConceptRows(sourceId int, cohortId int, conceptIds []int, breakdownConceptId int, sortedConcepts []string) ([][]string, error) {
+	conceptIdToName := make(map[int]string)
+	conceptInformations, err := u.conceptModel.RetrieveInfoBySourceIdAndConceptIds(sourceId, conceptIds)
+	if err != nil {
+		return nil, fmt.Errorf("Could not retrieve concept informations due to error: %s", err.Error())
+	}
+	for _, conceptInformation := range conceptInformations {
+		conceptIdToName[conceptInformation.ConceptId] = conceptInformation.ConceptName
+	}
+
+	var rows [][]string
+	for idx, conceptId := range conceptIds {
+		filterConceptIds := conceptIds[0 : idx+1]
+		breakdownStats, err := u.conceptModel.RetrieveBreakdownStatsBySourceIdAndCohortIdAndConceptIds(sourceId, cohortId, filterConceptIds, breakdownConceptId)
+		if err != nil {
+			return nil, fmt.Errorf("Could not retrieve concept Breakdown for concepts %v due to error: %s", filterConceptIds, err.Error())
+		}
+
+		conceptValuesToPeopleCount := getConceptValueToPeopleCount(breakdownStats)
+		rows = append(rows, generateConceptRow(conceptIdToName[conceptId], conceptValuesToPeopleCount, sortedConcepts))
+	}
+
+	return rows, nil
+}
+
+func (u ConceptController) RetrieveAttritionTable(c *gin.Context) {
+	sourceId, cohortId, conceptIds, err1 := parseSourceIdAndCohortIdAndConceptIds(c)
+	breakdownConceptId, err2 := utils.ParseNumericArg(c, "breakdownconceptid")
+
+	if err1 == nil && err2 == nil {
+		breakdownStats, err := u.conceptModel.RetrieveBreakdownStatsBySourceIdAndCohortId(sourceId, cohortId, breakdownConceptId)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"message": "Error retrieving breakdown stats", "error": err})
+			c.Abort()
+			return
+		}
+		conceptValuesToPeopleCount := getConceptValueToPeopleCount(breakdownStats)
+
+		cohortName, err := getCohortName(cohortId)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"message": "Error retrieving cohort name", "error": err})
+			c.Abort()
+			return
+		}
+
+		headerAndNonFilteredRow := genereateHeaderAndNonFilteredRow(cohortName, conceptValuesToPeopleCount)
+		sortedConcepts := getSortedConcepts(conceptValuesToPeopleCount)
+		filteredRows, err := u.getFilteredConceptRows(sourceId, cohortId, conceptIds, breakdownConceptId, sortedConcepts)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"message": "Error retrieving concept breakdown with filtered conceptIds", "error": err})
+			c.Abort()
+			return
+		}
+
+		b := GenerateAttritionCSV(headerAndNonFilteredRow, filteredRows)
+		c.String(http.StatusOK, b.String())
+		return
+	}
+
+	c.JSON(http.StatusBadRequest, gin.H{"message": "bad request"})
+	c.Abort()
+}
+
+func getSortedConcepts(conceptToPeopleCount map[string]int) []string {
```

Comment:
continuing on comment above: also rename `conceptToPeopleCount` to the clearer/more explicit name `conceptValuesToPeopleCount` (which is also consistent with the name used elsewhere in the code)